### PR TITLE
Add test for ping_agent run session

### PR DESCRIPTION
**Pull request by Acuadros95**

### CODE REVIEW

File: test/test/publisher_subscriber/PublisherSubscriberInteraction.cpp [hunk 1]
```diff
@@ -173,6 +173,17 @@ TEST_P(PublisherSubscriberNoLost, PubSub1ContinousFragmentedTopic)
     publisher_.publish(1, 0x80, 1, message);
 }
 
+TEST_P(PublisherSubscriberNoLost, PubSub1WithPing)
+{
+    size_t message_number = 1;
+    publisher_.publish(1, 0x80, message_number, SMALL_MESSAGE);
+    subscriber_.request_data(1, 0x80, SMALL_MESSAGE);
+    subscriber_.ping_agent(transport_);
```
Comment (Acuadros95) [inline test/test/publisher_subscriber/PublisherSubscriberInteraction.cpp:181]: Should we ping multiple times to get a topic?

**Comment by Acuadros95** [comment]
Set client branch to https://github.com/eProsima/Micro-XRCE-DDS-Client/pull/289 once regression test fails.

File: test/test/publisher_subscriber/PublisherSubscriberInteraction.cpp (hunk 1 above)
Comment (pablogs9) [reply test/test/publisher_subscriber/PublisherSubscriberInteraction.cpp:181]: It depends on when the pong arrives. Maybe if pong arrives before the topic, the topic is not processed

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Linux/290/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Windows/310/)

File: CMakeLists.txt [hunk 2]
```diff
@@ -29,7 +29,7 @@ endif()
 
 project(microxrcedds VERSION "2.0.0")
 
-set(_client_tag develop)
+set(_client_tag fix/session_ping)
```
Comment (Acuadros95) [inline CMakeLists.txt:32]:
```suggestion
set(_client_tag develop)
```
Before merge

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Linux/291/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Windows/311/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Linux/292/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Windows/312/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Linux/293/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Windows/313/)

**Review by pablogs9 — APPROVED** [review]

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Linux/295/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Windows/315/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Linux/296/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Windows/316/)

File: test/test/publisher_subscriber/PublisherSubscriberInteraction.cpp [hunk 3]
```diff
@@ -173,6 +173,21 @@ TEST_P(PublisherSubscriberNoLost, PubSub1ContinousFragmentedTopic)
     publisher_.publish(1, 0x80, 1, message);
 }
 
+TEST_P(PublisherSubscriberNoLost, PubSub1WithPing)
+{
+    size_t message_number = 1;
+    publisher_.publish(1, 0x80, message_number, SMALL_MESSAGE);
+    subscriber_.request_data(1, 0x80, SMALL_MESSAGE);
+
+    for (size_t i = 0; i < 5; i++)
```
Comment (pablogs9) [inline test/test/publisher_subscriber/PublisherSubscriberInteraction.cpp:182]: As we know that the agent will transmit the topic to the client at some moment in between of the ping answers due to the output message queue, what do you think about doing something like:

```c
int64_t start_time = uxr_millis();

while(uxr_millis() -  start_time < 10000 && subscriber_.get_received_topics() != message_number){
        subscriber_.ping_agent_session();
}

ASSERT_EQ(subscriber_.get_received_topics(), message_number);
```

Comment (Acuadros95) [reply test/test/publisher_subscriber/PublisherSubscriberInteraction.cpp]: Updated 57e4658c53749a89e5477488e52c3a3771aeab12

Comment (pablogs9) [reply test/test/publisher_subscriber/PublisherSubscriberInteraction.cpp]: Cool

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Linux/297/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Windows/317/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Linux/298/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Windows/318/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Linux/299/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Windows/319/)

**Review by pablogs9 — APPROVED** [review]

**Pull request merged**